js/os_test.ts

```diff
@@ -30,6 +30,83 @@ test(function envFailure(): void {
   assert(caughtError);
 });
 
+test(function envFailure(): void {
+  let caughtError = false;
+  try {
+    Deno.env();
+  } catch (err) {
+    caughtError = true;
+    assertEquals(err.kind, Deno.ErrorKind.PermissionDenied);
+    assertEquals(err.name, "PermissionDenied");
+  }
+
+  assert(caughtError);
+});
+
+if (Deno.build.os === "win") {
+  testPerm({ env: true, run: true }, async function envCaseInsensitive() {
+    // This test verifies that on Windows, environment variables are
+    // case-insensitive. Case normalization needs be done using the collation
+    // that Windows uses, rather than naively using String.toLowerCase().
+
+    assertEquals(Deno.env("path"), Deno.env("PATH"));
+    assertEquals(Deno.env("Path"), Deno.env("PATH"));
+
+    // Utility function that runs a Deno subprocess with the environment
+    // specified in `env`. The subprocess reads the environment variables
+    // specified in `keys` and outputs them as a JSON object.
+    const runGetEnv = async (env, keys) => {
+      const src = `
+        Deno.stdout.write((new TextEncoder()).encode(JSON.stringify(
+          ${JSON.stringify(keys)}.map(k => Deno.env(k))
+        )))`;
```

Comment:
How about this:

`console.log(JSON.stringify(${JSON.stringify(keys)}.map(k => Deno.env(k))))`

Reply:
Yeah better. The `Deno.stdout.write` things was a relic from an early version of the test, where I wasn't using JSON output yet.